src/main/java/com/p6spy/engine/common/ConnectionInformation.java

```diff
@@ -219,4 +220,18 @@ public void setTimeToGetConnectionNs(long timeToGetConnectionNs) {
   public ConnectionInformation getConnectionInformation() {
     return this;
   }
+  
+  public void setUrl(String url) {
+    this.url = url;
+  }
+
+  /**
+   * Returns the {@link #url} which was used to create the {@link #connection}
```

Comment:
I'd make it even more clear that it only returns a non-null value, if the connection was obtained by `Driver#connect`:
```
/**
 * If the connection was created by calling {@link Driver#connect(String, Properties)},
 * returns the {@link #url} which was used to create the {@link #connection}.
 * Returns {@code null} otherwise.
 *
 * @return the {@link #url} used to create the {@link #connection}, may be {@code null}
 */
```